src/main/java/frc/robot/vision/filters/Filter.java

```diff
@@ -0,0 +1,45 @@
+package frc.robot.vision.filters;
+
+import frc.robot.vision.rawdata.RawVisionData;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class Filter<T extends RawVisionData> {
+
+	private final Function<T, Boolean> filteringFunction;
+
+	public Filter(Function<T, Boolean> filteringFunction) {
+		this.filteringFunction = filteringFunction;
+	}
+
+	public boolean doesFilterPass(T data) {
+		return filteringFunction.apply(data);
+	}
+
+	public Filter<T> andThen(Filter<T> anotherFilter) {
```

Comment:
its not andThen its and or something like it

Reply:
it is andThen cos it needs both to be true